optimizely/project_config.py

```diff
@@ -134,11 +135,45 @@ def __init__(self, datafile, logger, error_handler):
         self.experiment_feature_map = {}
         for feature in self.feature_key_map.values():
             feature.variables = self._generate_key_map(feature.variables, 'key', entities.Variable)
-
             for exp_id in feature.experimentIds:
                 # Add this experiment in experiment-feature map.
                 self.experiment_feature_map[exp_id] = [feature.id]
 
+        # all rules(experiment rules and delivery rules) for each flag
+        self.flag_rules_map = {}
+        for flag in self.feature_flags:
+
+            experiments = []
+            if not flag['experimentIds'] == '':
+                for exp_id in flag['experimentIds']:
+                    experiments.append(self.experiment_id_map[exp_id])
+            if not flag['rolloutId'] == '':
```

Comment:
check this as well.
https://github.com/optimizely/python-sdk/blob/master/optimizely/project_config.py#L84

Reply:
in progress

Reply:
Not refactored, didn't see how to optimize it.

Reply:
may be file a ticket, need to refactor and assign to me, i will check it out.